meteor/client/ui/Settings/ShowStyle/VariantListItem.tsx

```diff
@@ -95,6 +101,83 @@ export const VariantListItem: React.FunctionComponent<IShowStyleVariantItemProps
 		})
 	}
 
+	function showStyleVariantNameAlreadyExists(showStyleVariant: ShowStyleVariant): boolean {
+		return props.dragDropVariants.some(
+			(variant: ShowStyleVariant) => variant.name === showStyleVariant.name && variant._id !== showStyleVariant._id
+		)
+	}
+
+	function addDuplicationCountToName(showStyleVariant: ShowStyleVariant, duplicatorIncrement: number = 0): void {
+		showStyleVariant.name = `${getNameWithRemovedDuplicateIndicator(showStyleVariant)} (${
+			getDuplicatedShowStyleVariantCount(showStyleVariant) + duplicatorIncrement
+		})`
+		if (showStyleVariantNameAlreadyExists(showStyleVariant)) {
+			addDuplicationCountToName(showStyleVariant, duplicatorIncrement + 1)
+		} else {
+			MeteorCall.showstyles.renameShowStyleVariant(props.showStyleBase._id, props.showStyleVariant).catch(logger.warn)
+		}
+	}
+
+	function getDuplicatedShowStyleVariantCount(showStyleVariant: ShowStyleVariant): number {
+		const variantsWithName = props.dragDropVariants.filter((variant: ShowStyleVariant) => {
+			const variantNameWithoutDuplicator = getNameWithRemovedDuplicateIndicator(variant)
+			return variantNameWithoutDuplicator === showStyleVariant.name && variant._id !== showStyleVariant._id
+		})
+		return variantsWithName.length
+	}
+
+	function getNameWithRemovedDuplicateIndicator(showStyleVariant: ShowStyleVariant): string {
+		return showStyleVariant.name.replace(/ +\(\d+\)$/, '')
+	}
+
+	function replaceOtherShowStyleVariant(): void {
+		const variantToReplace = props.dragDropVariants.find((variant: ShowStyleVariant) => {
+			return props.showStyleVariant.name === variant.name && props.showStyleVariant._id !== variant._id
+		})
+		if (!variantToReplace) {
+			return
+		}
+		MeteorCall.showstyles.removeShowStyleVariant(variantToReplace._id).catch(logger.warn)
```

Comment:
I noticed that when replacing a variant, the replacement gets moved to the end. It would seem more logical for the variant to stay in place, but looks like it would need some extra work and possibly some new methods in the API to solve that with the way the variants are imported and reordered, so I'm not saying it's a deal breaker

Reply:
When replacing a variant due to the changes given in this specific PR, meaning whenever a name is edited to be a duplicated, it will stay in place when replacing. When replacing on import the behaviour is as you say. I'll have a quick look at it before merging!

Reply:
Actually, when replacing on the duplication from name editing, it will remove the existing variant and keep the new one where you are editing it. I think that is intuitive. But the opposite would happen if i "steal" the rank from the original in the case of importing and replacing. I'm not sure what is best.

Reply:
So after a discussion with myself in this comment section and a real one with the other Rasmus. I've decided that compared to how big a task it is in regards to the scope of this specific PR, as well as the fact that it might be intuitive as is, i will keep the current implementation.